app/controllers/response_templates_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ResponseTemplatesController < ApplicationController
+  after_action :verify_authorized, except: %i[index]
+
+  def index
+    not_found unless current_user
+    @response_templates = if params[:type_of] && params[:personal_included] != "true"
+                          result = ResponseTemplate.where(user_id: nil, type_of: params[:type_of])
+                          handle_authorization(result)
+                          result
+                        elsif params[:type_of] == "mod_comment" && params[:personal_included] == "true"
+                          result = ResponseTemplate.
+                            where(user_id: nil, type_of: "mod_comment").
+                            union(user_id: current_user.id, type_of: "personal_comment")
+                          handle_authorization(result)
+                          result
+                        else
+                          skip_authorization
+                          ResponseTemplate.where(user_id: current_user.id)
+                        end
+  end
+
+  def create
+    authorize ResponseTemplate
+    @response_template = ResponseTemplate.new(permitted_attributes(ResponseTemplate))
+    @response_template.user_id = current_user.id
+    @response_template.content_type = "body_markdown"
+    @response_template.type_of = "personal_comment"
+
+    if @response_template.save
+      flash[:settings_notice] = "Your response template \"#{@response_template.title}\" was created."
+    else
+      flash[:error] = "Response template error: #{@response_template.errors.full_messages.to_sentence}"
+    end
+
+    redirect_back(fallback_location: user_settings_path)
+  end
+
+  def destroy
+    @response_template = ResponseTemplate.find(params[:id])
```

Comment:
Any reason to not memoize the `@response_template`?

Reply:
Just that I didn't think of it! I'm not 100% clear of when to memoize though; should this be memoize in the other actions as well as `destroy`?

Reply:
Looks like we have `@response_template` defined in `create`, `destroy, and `update`. The `create` method is a little different, but you could check for the `params[:id]` and pull it out into something like this:

```
def response_template
  @response_template ||= if params[:id].present?
    ResponseTemplate.find(params[:id])
  else
    ResponseTemplate.new(permitted_attributes(ResponseTemplate))
end
```

Not sure if this _exact_ code is actually right, but that's what I could imagine. Basically, we only want to memoize this if we know that it's not going to change, and it seems like, we're either `find`ing or creating a `new` response template, so this should be okay I think 🤔

Reply:
Ohh interesting, thanks for that! That makes a lot more sense. Will add it in!